src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -628,3 +631,79 @@ def _default_openai_function(
             "required": required,
         },
     }
+
+
+class RunEvalsPayload(TypedDict):
+    evaluator: Evaluator
+    record: Record
+    row_index: RowIndex
+
+
+def run_evals(
+    dataframe: DataFrame,
+    evaluators: List[Evaluator],
+    concurrency: int = 20,
+) -> DataFrame:
+    """
+    Applies a list of evaluators to every row of a dataframe. Outputs a
+    dataframe where each column corresponds to an evaluator and each row
+    corresponds to a row in the input dataframe.
+
+    Args:
+        dataframe (pd.DataFrame): A pandas dataframe in which each row
+        represents a record to be evaluated. All template variable names must
+        appear as column names in the dataframe (extra columns unrelated to the
+        template are permitted).
+
+        evaluators (List[Evaluator]): A list of evaluators with unique names.
+
+        concurrency (int, optional): An optional concurrency parameter. Defaults
+        to 20.
+
+    Returns:
+        DataFrame: A dataframe where each row contains the outputs of the
+        evaluators applied to the corresponding row of the input dataframe and
+        the column names match the names of the evaluators. The index of the
+        dataframe is the same as the index of the input dataframe.
+    """
+    if len(set(evaluator.name for evaluator in evaluators)) != len(evaluators):
+        raise ValueError("Evaluators must have unique names.")
+
+    async def _run_eval_async(
+        payload: RunEvalsPayload
+    ) -> Tuple[RowIndex, EvalName, EvaluationResult]:
+        row_index = payload["row_index"]
+        evaluator = payload["evaluator"]
+        record = payload["record"]
+        eval_result = await evaluator.aevaluate(record)
+        return row_index, evaluator.name, eval_result
+
+    def _run_eval_sync(payload: RunEvalsPayload) -> Tuple[RowIndex, EvalName, EvaluationResult]:
+        row_index = payload["row_index"]
+        evaluator = payload["evaluator"]
+        record = payload["record"]
+        eval_result = evaluator.evaluate(record)
+        return row_index, evaluator.name, eval_result
+
+    executor = get_executor_on_sync_context(
+        _run_eval_sync,
+        _run_eval_async,
+        concurrency=concurrency,
+        tqdm_bar_format=get_tqdm_progress_bar_formatter("llm_classify"),
+        exit_on_error=True,
+        fallback_return_value=(None, None),
+    )
+    payloads = [
+        {
+            "row_index": row_index,
+            "evaluator": evaluator,
+            "record": row.to_dict(),
+        }
+        for row_index, row in dataframe.iterrows()
+        for evaluator in evaluators
+    ]
+    results: DefaultDict[RowIndex, Dict[EvalName, EvalPrediction]] = defaultdict(dict)
+    for row_index, eval_name, eval_result in executor.run(payloads):
+        results[row_index][eval_name] = eval_result.prediction
+    index, data = zip(*results.items())
```

Comment:
Since `EvaluationResult` has been made into a class, it suggests that we want to eventually turn it into a compound struct instead of just a simple string. For example, what if the result consist of a triplet: score, label, and explanation, or what if some evaluators return labels, while others return numbers? How are we going to handle those use cases?

Reply:
The intent of `EvaluationResult` was to store those additional outputs in a single output, although there is not an explicit need for it in this PR.

Reply:
I think it's premature to introduce this abstraction. Removed in c3870be8.